suffuse/inode_gen.go

```diff
@@ -0,0 +1,38 @@
+package suffuse
+
+type InodeGen struct {
+  fresh chan InodeNum
+}
+
+func NewRootInode(startInode uint64)*Inode {
+  ichan := make(chan InodeNum)
+  // Monotonically increments the counter without any
+  // effort to track or reuse.
+  go func() {
+    count := uint64(startInode)
+    for {
+      ichan <- InodeNum(count)
+      count += 1
+    }
+  }()
+
+  gen := InodeGen { ichan }
+  return gen.NextDir()
+}
+
+func (x *InodeGen) Next(tp InodeType) *Inode {
```

Comment:
The name `Next` should in my opinion be `New`. The fact that it uses an incremental number internally does not need to be reflected in the name. I think `NewFile`, `NewDir`, ... is more intuitive.